### DISCUSSION TITLE
How to access inferred path types

### DISCUSSION THREAD

**Discussion by TheBestMoshe** [trpc/trpc#1311] [Q&A] [answered]
I am trying to make a generic table component, which will allow the user to pass in a trpc query path, and it will render the response in a table. 

I want Typescript to enforce the routes accepted by trpc.

Looking in the [source code](https://github.com/trpc/trpc/blob/f7cb49f8c741e80b4af96b3fb4947eb87e73c913/packages/react/src/createReactQueryHooks.tsx#L101), I see the `createReactQueryHooks` function has a type `TQueryValues` which I believe has the types I need. How can I access these types from within my code?

**Comment by TheBestMoshe** [comment, marked as answer]
Answering my own question, the argument types can be inferred using standard TypeScript:

```
 type PathTypes = Parameters<typeof trpc.useQuery>[0][0]
```

[StackOverflow](https://stackoverflow.com/a/51851844/9642772)

**Reply by KATT** [reply]
Awesome you solved it! You can also do `keyof AppRouter['_def']['queries']`
